hr_period/models/hr_fiscal_year.py

```diff
@@ -0,0 +1,324 @@
+    'bi-weekly': (relativedelta(weeks=2), 26),
+    'weekly': (relativedelta(weeks=1), 52),
+    'daily': (relativedelta(days=1), 365),
+}
+
+
+@api.model
+def get_schedules(self):
+    return [
+        ('annually', _('Annually (1)')),
+        ('semi-annually', _('Semi-annually (2)')),
+        ('quarterly', _('Quarterly (4)')),
+        ('bi-monthly', _('Bi-monthly (6)')),
+        ('monthly', _('Monthly (12)')),
+        ('semi-monthly', _('Semi-monthly (24)')),
+        ('bi-weekly', _('Bi-weekly (26)')),
+        ('weekly', _('Weekly (52)')),
+        ('daily', _('Daily (365)')),
+    ]
+
+
+@api.model
+def get_payment_days(self):
+    expr = _('%s day of the next period')
+    expr_2 = _('%s day of the current period')
+    return [
+        ('1', expr % _('First')),
+        ('2', expr % _('Second')),
+        ('3', expr % _('Third')),
+        ('4', expr % _('Fourth')),
+        ('5', expr % _('Fifth')),
+        ('0', expr_2 % _('Last')),
+    ]
+
+
+class HrFiscalYear(models.Model):
+    _name = 'hr.fiscalyear'
+    _description = 'HR Fiscal Year'
+
+    @api.model
+    def _default_date_start(self):
+        today = datetime.now()
+        return datetime(today.year, 1, 1).strftime(DF)
+
+    @api.model
+    def _default_date_stop(self):
+        today = datetime.now()
+        return datetime(today.year, 12, 31).strftime(DF)
+
+    name = fields.Char(
+        'Fiscal Year',
+        required=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    company_id = fields.Many2one(
+        'res.company',
+        'Company',
+        required=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        default=lambda obj: obj.env.user.company_id
+    )
+    date_start = fields.Date(
+        'Start Date',
+        required=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        help="The first day of the first period of the fiscal year.",
+        default=_default_date_start
+    )
+    date_stop = fields.Date(
+        'End Date',
+        required=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        help="The last day of the last period of the fiscal year.",
+        default=_default_date_stop
+    )
+    period_ids = fields.One2many(
+        'hr.period',
+        'fiscalyear_id',
+        'Periods',
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    state = fields.Selection(
+        [
+            ('draft', 'Draft'),
+            ('open', 'Open'),
+            ('done', 'Closed'),
+        ],
+        'Status',
+        readonly=True,
+        default='draft'
+    )
+    schedule_pay = fields.Selection(
+        get_schedules,
+        'Scheduled Pay',
+        required=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        default='monthly'
+    )
+    type_id = fields.Many2one(
+        'date.range.type',
```

Comment:
Please add a description to this field so it does not look ugly:

![image](https://user-images.githubusercontent.com/19620251/30956960-1ac8cf54-a439-11e7-9724-0205df137ee9.png)